libraries/c_sdk/standard/mqtt/src/private/iot_mqtt_internal.h

```diff
@@ -913,4 +735,178 @@ bool _IotMqtt_GetNextByte( void * pNetworkConnection,
 void _IotMqtt_CloseNetworkConnection( IotMqttDisconnectReason_t disconnectReason,
                                       _mqttConnection_t * pMqttConnection );
 
+/*----------------- MQTT Serailization /Deserialization Wrapper functions for Shim------------------*/
+
+/**
+ * @brief Generate a CONNECT packet from the given parameters by using MQTT v4 beta 2 serializer.
```

Comment:
v4 beta 2 may not be the official name of the library, but we can clean it up later once we have a finalized name.